File: calc-business-hours.py
import datetime


def is_leap_year(year):
    return True if year % 4 == 0 else False


def number_of_days(datetime):
    this_month = datetime.month
    monthes = {"with31Days": [1, 3, 5, 7, 8, 10, 12], "with30Days": [4, 6, 9, 11]}

    if this_month in monthes["with31Days"]:
        return 31
    elif this_month in monthes["with30Days"]:
        return 30
    elif is_leap_year(datetime.year):
        return 29
    else:
        return 28


now = datetime.datetime.now()

work_hours = int(input("Insira quantidade de horas trabalhadas por dia: "))
ignore_hours = int(input("Horas de abono (feriados, folgas, short day): "))
exclude_days = ignore_hours // work_hours
exclude_remain = ignore_hours % work_hours
business_days = 0

for day in range(number_of_days(now)):
    date = datetime.date(now.year, now.month, day + 1)
    week_day = date.weekday()
    # 5 = sábado, 6 = domingo
    business_days += 0 if week_day in [5, 6] else 1

print(
    f"No mês {now.month}/{now.year} tiveram {business_days} dias úteis.\n",
    f"Tempo total a ser trabalhado: {business_days * work_hours - ignore_hours}\n",
    f"Tempo total excluindo abonos: {business_days * work_hours}\n",
    f"Qtd. de dias úteis trabalhados incluindo horas de abono:",
    f"{business_days - exclude_days} dias úteis",
    f"com {exclude_remain} horas restantes de abono",
)

File: calc-business-hours.js
const workHours = 8;
let businessDays = 0;
const daysToIgnore = 2;

const today = new Date();
const thisMonth = today.getMonth() + 1;
const thisYear = today.getFullYear();

for (
  let dayOfTheMonth = 1;
  dayOfTheMonth <= totalMonthDays(today);
  dayOfTheMonth++
) {
  const thisDate = new Date(`${thisYear}-${thisMonth}-${dayOfTheMonth}`);
  // [domingo: 0, sábado: 6]
  businessDays = [0, 6].includes(thisDate.getDay())
    ? businessDays
    : businessDays + 1;
}

businessDays = businessDays - daysToIgnore;

console.log(
  `O mês ${thisMonth} de ${thisYear} tem ${businessDays} dias úteis, totalizando ${
    businessDays * workHours
  } horas trabalhadas`,
);

const isLeapYear = (year) => (year % 4 === 0 ? true : false);

function totalMonthDays(today) {
  const monthes = {
    with31Days: [0, 2, 4, 6, 7, 9, 11],
    with30Days: [3, 5, 8, 10],
  };
  const thisMonth = today.getMonth();
  return monthes.with31Days.includes(thisMonth)
    ? 31
    : monthes.with30Days.includes(thisMonth)
    ? 30
    : isLeapYear(today.getFullYear())
    ? 29
    : 28;
}
